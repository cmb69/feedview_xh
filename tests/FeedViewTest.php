<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Feedview_XH.
 *
 * Feedview_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Feedview_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Feedview_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Feedview;

use ApprovalTests\Approvals;
use Feedview\Infra\FakeFeedReader;
use Feedview\Infra\View;
use PHPUnit\Framework\TestCase;

class FeedViewTest extends TestCase
{
    public function testRendersFeedWithoutItems(): void
    {
        global $pth;
        $pth = ["folder" => ["plugins" => ""]];
        $conf = XH_includeVar("./config/config.php", "plugin_cf")["feedview"];
        $text = XH_includeVar("./languages/en.php", "plugin_tx")["feedview"];
        $sut = new FeedView($conf, $text, new FakeFeedReader, new View("./views/", $text));
        $response = $sut("irrelevant_url", "default");
        Approvals::verifyHtml($response);
    }
}
