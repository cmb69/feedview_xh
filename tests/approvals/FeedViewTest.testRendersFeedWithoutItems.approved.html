<!-- Feedview_XH: default feed view -->
<div class="feedview_header">
  <h4 class="feedview_feed_title">
    <a href="http://example.com/feed" target="_blank">
      The best Newsfeed!    </a>
  </h4>
  <div class="feedview_feed_description">The most recent news are available only here.</div>
</div>
